src/objective.jl

```diff
@@ -78,3 +78,48 @@ function set_objective(model::Model, sense::MOI.OptimizationSense,
     set_objective_sense(model, sense)
     set_objective_function(model, func)
 end
+
+"""
+    objective_function(m::Model, T::Type{<:AbstractJuMPScalar})
+
+Return a object of type `T` representing the objective function.
+Error if the objective is not convertible to type `T`.
+
+## Examples
+
+```jldoctest; setup = :(using JuMP)
+julia> model = Model()
+A JuMP Model
+
+julia> @variable(model, x)
+x
+
+julia> @objective(model, Min, 2x + 1)
+2 x + 1
+
+julia> JuMP.objective_function(model, JuMP.GenericAffExpr{Float64})
```

Comment:
Most users are probably interested in `AffExpr` instead of `GenericAffExpr{Float64}`.